libminifi/include/core/logging/internal/LogCompressorSink.h

```diff
@@ -54,7 +48,7 @@ class LogCompressorSink : public spdlog::sinks::base_sink<std::mutex> {
   void flush_() override;
 
  public:
-  explicit LogCompressorSink(LogQueueSize cache_size, LogQueueSize compressed_size, std::shared_ptr<logging::Logger> logger);
+  LogCompressorSink(LogQueueSize cache_size, LogQueueSize compressed_size, const std::shared_ptr<logging::Logger>& logger);
```

Comment:
The detail you are missing is that the `logger` parameter is used twice in the constructor: it is passed to the `ActiveCompressor::Allocator` in the `compressed_logs_` field, as well as stored directly in the `compressor_logger_` field.  I did not like the looks of passing by value, copying once and then moving.  I can change it to that if you prefer it.

I removed `explicit` because it doesn't make much sense for constructors with multiple parameters, but I can put it back if you think it's useful.

Reply:
Makes sense. It's fine as is.